packages/webdriverio/src/commands/element/getValue.ts

```diff
@@ -20,11 +22,13 @@
  * @uses protocol/elements, protocol/elementIdProperty
  *
  */
-export function getValue (this: WebdriverIO.Element) {
+export function getValue<T>(this: WebdriverIO.Element): Promise<T>
+export function getValue(this: WebdriverIO.Element): Promise<string>
+export function getValue<T>(this: WebdriverIO.Element) {
```

Comment:
I think the above overwrites:
```ts
export function getValue<T>(this: WebdriverIO.Element): Promise<T>
export function getValue(this: WebdriverIO.Element): Promise<string>
```
don't make a difference because TypeScript won't be able to resolve the types pre-runtime. I would suggest to default to `string` but allow users to pass in a type when they do mobile native testing.
```suggestion
export function getValue<T = string>(this: WebdriverIO.Element) {
```

Reply:
You are right, I stashed some code and this accidentally seems to be the old code that I had instead of what I intended. I'll update this thanks!

Reply:
I double checked this and it's actually my intention to create a TypeScript function overload, this doesn't work?

I'm not sure what you mean by pre-runtime but I overwrote the type in the node_modules in one of my projects and that did seem to work.

Reference: https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads

Reply:
Overloading the function doesn't have an effect here because TypeScript doesn't know if the browser instance variable is running a browser session or a native mobile session. TypeScript can not resolve types at runtime when we know more about these variables. If you could create some type tests you will see that you always get the same return type from TypeScript.